utils/priority_mutex_test.go

```diff
@@ -16,17 +16,41 @@ package utils
 
 import (
 	"context"
+	"sync"
 	"testing"
-	"time"
 
 	"github.com/stretchr/testify/assert"
 	"golang.org/x/sync/errgroup"
 )
 
+// A wrapper around the PriorityMutex being tested to properly assert invariants without the
+// possibility of race conditions. The `lockWg` completes after the low/high queue is appended to
+// but before the blocking `wait` call so we can assert invariants on the size of the queue.
+type wrappedPriorityMutex struct {
+	*PriorityMutex
+	lockWg sync.WaitGroup
+}
+
+func newWrappedPriorityMutex(numLockCalls int) *wrappedPriorityMutex {
+	w := &wrappedPriorityMutex{PriorityMutex: new(PriorityMutex)}
+	w.lockWg.Add(numLockCalls)
+	return w
+}
+
+func (w *wrappedPriorityMutex) Lock(priority bool) {
+	c := w.PriorityMutex.lockInternal(priority)
+	w.lockWg.Done()
+	w.wait(c)
```

Comment:
In this method, if we do following, I think we can initialize the WaitGroup counter with 60 instead of 61. 60 will make the test easier to understand. lmk what you think. 🙏 
```
func (w *wrappedPriorityMutex) Lock(priority bool) {
	c := w.PriorityMutex.lockInternal(priority)
        if c != nil {
          w.lockWg.Done()
	  w.wait(c)
        }
}
```